src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -358,7 +344,17 @@ public PackageSource GetPackageSourceBySource(string source)
                 throw new ArgumentException(Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(source));
             }
 
-            return GetPackageSource(source, LoadPackageSourceLookupBySource());
+            List<PackageSource> packageSources = LoadPackageSources(Settings, _configurationDefaultSources);
+
+            foreach (var packageSource in packageSources)
+            {
+                if (string.Equals(source, packageSource.Source, StringComparison.OrdinalIgnoreCase))
+                {
+                    return packageSource;
+                }
+            }
+
+            return null;
```

Comment:
We could simplify this logic using LINQ. Is the original implementation chosen to avoid memory allocations? For instance, the code can be condensed to:

```cs
return packageSources.FirstOrDefault(ps => string.Equals(source, ps.Source, StringComparison.OrdinalIgnoreCase));
```